### ISSUE TITLE
fails on CentOS 7 and 8

### ISSUE THREAD

**Issue by astlaurent** [checktheroads/hyperglass#31]
Tried installing this on CentOS 7 or 8 as that is our standard linux OS. Not a huge deal, I can run it on ubuntu for a while till things shake out, just wanted you to know i ran into some issues. 

If fails a number of items along the way. First it fails building uvloop and it requires the python devel tools. yum install python36-devel. That moved past that problem. The next problem i ran into is the service refuses to install. 

The unit files have no [Install] section. They are not meant to be enabled
using systemctl.
Possible reasons for having this kind of units are:
1) A unit may be statically enabled by being symlinked from another unit's
   .wants/ or .requires/ directory.
2) A unit's purpose may be to act as a helper for some other unit which has
   a requirement dependency on it.
3) A unit may be started when needed via activation (socket, path, timer,
   D-Bus, udev, scripted systemctl call, ...).

Then if i just attempt to start it using the config files from my ubuntu test it throws errors.

[ERROR] 20200410 15:31:25 | hyperglass.configuration.models._utils:161 | validate_image → /etc/hyperglass/static/images/hyperglass-light.png
[ERROR] 20200410 15:31:25 | hyperglass.configuration.models._utils:161 | validate_image → /etc/hyperglass/static/images/hyperglass-dark.png
[ERROR] 20200410 15:31:25 | hyperglass.configuration.models._utils:161 | validate_image → /etc/hyperglass/static/images/hyperglass-opengraph.png
[INFO] 20200410 15:31:25 | hyperglass.configuration:38 | <module> → Configuration directory: /etc/hyperglass
[DEBUG] 20200410 15:31:25 | hyperglass.configuration:113 | _set_log_level → Debugging enabled
[DEBUG] 20200410 15:31:25 | hyperglass.configuration:148 | _config_commands → Unvalidated commands: {'cisco_xr': {'ipv4_default': {'bgp_aspath': "show bgp ipv4 unicast regexp {target} | utility egrep -v '\\\\(BGP |Table |Non-stop\\\\)'", 'bgp_community': "show bgp ipv4 unicast community {target} | utility egrep -v '\\\\(BGP |Table |Non-stop\\\\)'", 'bgp_route': "show bgp ipv4 unicast {target} | utility egrep '\\\\(BGP routing table entry|Path \\\\#|aggregated by|Origin |Community:|validity|from\\\\)'", 'ping': 'ping ipv4 {target} count 5', 'traceroute': 'traceroute ipv4 {target} timeout 1 probe 2'}, 'ipv4_vpn': {'bgp_aspath': 'show bgp vpnv4 unicast vrf {vrf} regexp {target} | utility egrep -v \\\\(BGP |Table |Non-stop\\\\)', 'bgp_community': 'show bgp vpnv4 unicast vrf {vrf} community {target} | utility egrep -v \\\\(BGP |Table |Non-stop\\\\)', 'bgp_route': 'show bgp vpnv4 unicast vrf {vrf} {target} | util egrep \\\\(BGP routing table entry|Path \\\\#|aggregated by|Origin |Community:|validity| from \\\\)', 'ping': 'ping vrf {vrf} {target} count 5 source {source}', 'traceroute': 'traceroute vrf {vrf} {target} timeout 1 probe 2 source {source}'}, 'ipv6_default': {'bgp_aspath': "show bgp ipv6 unicast regexp {target} | utility egrep -v '\\\\(BGP |Table |Non-stop\\\\)'", 'bgp_community': "show bgp ipv6 unicast community {target} | utility egrep -v '\\\\(BGP |Table |Non-stop\\\\)'", 'bgp_route': "show bgp ipv6 unicast {target} | utility egrep '\\(BGP routing table entry|Path \\#|aggregated by|Origin |Community:|validity|from\\)'", 'ping': 'ping ipv6 {target} count 5', 'traceroute': 'traceroute ipv6 {target} timeout 1 probe 2'}, 'ipv6_vpn': {'bgp_aspath': 'show bgp vpnv6 unicast vrf {vrf} regexp {target} | utility egrep -v \\\\(BGP |Table |Non-stop\\\\)', 'bgp_community': 'show bgp vpnv6 unicast vrf {vrf} community {target} | utility egrep -v \\\\(BGP |Table |Non-stop\\\\)', 'bgp_route': 'show bgp vpnv6 unicast vrf {vrf} {target} | util egrep \\\\(BGP routing table entry|Path \\\\#|aggregated by|Origin |Community:|validity| from \\\\)', 'ping': 'ping vrf {vrf} {target} count 5 source {source}', 'traceroute': 'traceroute vrf {vrf} {target} timeout 1 probe 2 source {source}'}}}
[ERROR] 20200410 15:31:25 | hyperglass.exceptions:26 | __init__ → [WARNING]
Traceback (most recent call last):
  File "./hyperglass", line 11, in <module>
    sys.exit(CLI())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/hyperglass/cli/commands.py", line 81, in start
    from hyperglass.api import start, ASGI_PARAMS
  File "/usr/local/lib/python3.6/site-packages/hyperglass/api/__init__.py", line 19, in <module>
    from hyperglass.api.events import on_startup, on_shutdown
  File "/usr/local/lib/python3.6/site-packages/hyperglass/api/events.py", line 15, in <module>
    from hyperglass.configuration import (
  File "/usr/local/lib/python3.6/site-packages/hyperglass/configuration/__init__.py", line 180, in <module>
    _user_devices = aiorun(_config_devices())
  File "/usr/local/lib/python3.6/site-packages/hyperglass/compat.py", line 72, in _patched_run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/usr/local/lib/python3.6/site-packages/hyperglass/configuration/__init__.py", line 164, in _config_devices
    raise ConfigError(error_msg=str(yaml_error)) from None
hyperglass.exceptions.ConfigError

**Comment by checktheroads** [comment]
Hi,

Yeah, I haven’t had a chance to test v1 on CentOS yet. I know it works as one of the folks in the Gitter group was able to get through the issues, I do remember python3-devel being one of them.

I’ll get the install script fixed for CentOS/RHEL as soon as I can.

Thanks,
Matt

**Comment by astlaurent** [comment]
Hi

I was able to work through the issues on centos 7. Here is some stuff i had to do to help you with the docs

I followed your manual centos instructions. I had to do the following items as well before installing hyperglass
activate python 3.6 with scl enable rh-python36 bash. without this python will stay on version 2.x
install python3-devel

then install hyperglass via pip command

then enable and run redis-server
systemctl start redis
systemctl enable redis

Also disabled the built in firewall but users could allow the web port in as well
systemctl disable firewalld
